rr_common/src/image_pcl_converter/image_pcl_converter.cpp

```diff
@@ -17,34 +17,38 @@
 #include <boost/algorithm/string.hpp>
 
 using namespace std;
-using namespace cv;
-using namespace ros;
 
 using uchar = unsigned char;
 
-map<string, Publisher> cloud_pubs;
+map<string, ros::Publisher> cloud_pubs;
 double pxPerMeter;
 
-void transformedImageCB(const sensor_msgs::ImageConstPtr& msg, string topic) {
+pcl::PointCloud<pcl::PointXYZ>::Ptr cloud;
+
+void transformedImageCB(const sensor_msgs::ImageConstPtr& msg, const string& topic) {
     if(cloud_pubs[topic].getNumSubscribers() == 0) {
-        // no one is listening. go home and cry
         return;
     }
 
-    cv_bridge::CvImagePtr cv_ptr;
+    cv_bridge::CvImageConstPtr cv_ptr;
     try {
-        cv_ptr = cv_bridge::toCvCopy(msg, "bgr8");
+        cv_ptr = cv_bridge::toCvShare(msg);
     } catch(cv_bridge::Exception& e) {
         ROS_ERROR("CV_Bridge error: %s", e.what());
         return;
     }
 
-    pcl::PointCloud<pcl::PointXYZ>::Ptr cloud(new pcl::PointCloud<pcl::PointXYZ>);
-    Mat transformed = cv_ptr->image;
+    cloud->clear();
+    const cv::Mat& in_image = cv_ptr->image;
+
+    // get outline
+    cv::Mat transformed(in_image.size(), CV_8UC1);
+    cv::Canny(in_image, transformed, 10, 0);
```

Comment:
Nitpicking but since we have a binary image, canny takes more steps than needed. A simple Laplace should work just as well and do less work. Laplace will output gradients and not just an binary edge image, so the below check will need to account for negative numbers as well: `row[c] != 0;`

Reply:
Changing Canny to Laplacian does seem to speed up the inner pointcloud conversion loop, but it already took less than a millisecond anyway :man_shrugging: